devito/archinfo.py

```diff
@@ -60,18 +62,36 @@ def get_cpu_brand():
             except:
                 pass
 
-            return None
+            try:
+                return cpuinfo.get_cpu_info().get('raw_arch_string')
+            except:
+                pass
 
         cpu_info['flags'] = get_cpu_flags()
         cpu_info['brand'] = get_cpu_brand()
 
-    if not all(i in cpu_info for i in ('flags', 'brand')):
-        # Fallback
-        ci = cpuinfo.get_cpu_info()
-        cpu_info['flags'] = ci.get('flags')
-        cpu_info['brand'] = ci.get('brand')
+    try:
+        if 'flags' not in cpu_info:
```

Comment:
is it true that the only possibility for `flags` not to be in `cpu_info` is that `lines` is empty? because if we enter the `if lines:` at line 34, then we go through lines 70-71, hence `cpu_info['flags']` should be defined?

Reply:
This is the part where I do not know what happens in OSX, tests fail only there.